### PR TITLE
Top Level Waf Resource Swagger changes, Changes in PR#5906 by yanshu + some comments addressed

### PR DESCRIPTION
**Pull request by venk8**
…comments addressed on the right branch

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/webapplicationfirewall.json [hunk 1]
```diff
@@ -288,13 +288,21 @@
           "description": "Describes  policySettings for policy",
           "$ref": "#/definitions/PolicySettings"
         },
+        "policyDetailSettings": {
+          "description": "Describes  policyDetailSettings for policy",
```
Comment (EvgenyAgafonchikov) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/webapplicationfirewall.json]: minor: double space
```suggestion
          "description": "Describes policyDetailSettings for policy",
```

File: specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/webapplicationfirewall.json [hunk 2]
```diff
@@ -534,6 +567,115 @@
...
+      "selectorMatchOperator": {
+        "type": "string",
+        "description": "When matchVariable is a collection, operate on the selector to specify which elements in the collection this exclusion applies to."
+      },
+      "selector": {
+        "type": "string",
+        "description": "When matchVariable is a collection, operator used to specify which elements in the collection this exclusion applies to."
+      }
+    },
+    "required": [
```
Comment (EvgenyAgafonchikov) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/webapplicationfirewall.json]: Please check are all of these items required?
Say, for [ApplicationGatewayFirewallExclusion](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/applicationGateway.json#L2334) I was able to create it using matchVariable only (calling API directly).
I.e. I omitted `selectorMatchOperator` and `selector` both and it still worked well.
Is it also the case there?

File: specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/webapplicationfirewall.json [hunk 3]
```diff
@@ -105,7 +105,7 @@
         "x-ms-pageable": {
           "nextLinkName": "nextLink"
         }
-    }
+      }
     },
     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/ApplicationGatewayWebApplicationFirewallPolicies/{policyName}": {
```
Comment (EvgenyAgafonchikov) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/webapplicationfirewall.json]: to keep consistency between resources
```suggestion
     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/applicationGatewayWebApplicationFirewallPolicies/{policyName}": {
```

File: specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/webapplicationfirewall.json [hunk 4]
```diff
@@ -534,6 +567,115 @@
       }
     }
   },
+  "ManagedRuleSets": {
```
Comment (EvgenyAgafonchikov) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/webapplicationfirewall.json]: Please put these definitions (this and below until `parameters`) under the [definitions](https://github.com/Azure/azure-rest-api-specs/pull/5921/files#diff-a369a92061eba6d5f1ece36a6c9309a8R264) entry.

You could use [these validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) to check for structure issues locally.

File: specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/webapplicationfirewall.json [hunk 5]
```diff
@@ -375,6 +383,31 @@
         }
       }
     },
+    "policyDetailSettings": {
```
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/webapplicationfirewall.json]: Please start definition's name with capital letter for consistency

**Review by anton-evseev — CHANGES_REQUESTED** [review]

File: specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/webapplicationfirewall.json (hunk 2 above)
Comment (venkatsvpr) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/webapplicationfirewall.json]: Only matchVariable is required. There can be a case with selector and selectorMatchOperator as null. I will address the same.

File: specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/webapplicationfirewall.json (hunk 5 above)
Comment (venkatsvpr) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/webapplicationfirewall.json]: Sure. I will address the same.

**Review by dsgouda — CHANGES_REQUESTED** [review]

File: specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/webapplicationfirewall.json (hunk 4 above)
Comment (venkatsvpr) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/webapplicationfirewall.json]: Added it on to definitions.

**Review by dsgouda — APPROVED** [review]
LGTM

**Comment by asridharan** [comment]
@dsgouda this swagger shouldn't have been merged. We need to revert it since the NRP implementation and roll out is not complete yet. 

@venkatsvpr can you kindly follow up on this. This should not be merged.

**Comment by dsgouda** [comment]
@venkatsvpr Please open a PR to revert these changes. Kindly add [DO NOT MERGE] in the PR title so we don't merge it accidentally

**Comment by venkatsvpr** [comment]
@dsgouda  [PR:5971](https://github.com/Azure/azure-rest-api-specs/pull/5971)
this will revert the changes that were merged. I will send out another PR with the changes we need with the DO NOT MERGE titile.

**Comment by dsgouda** [comment]
@venkatsvpr just to be clear. I will have to merge the changes that are to revert these changes, the PR that you open for these changes in the future must have the [DO NOT MERGE] title

**Comment by venkatsvpr** [comment]
Yes.. That's what I am gonna do. Thanks for explaining!

**Comment by venkatsvpr** [comment]
Hey Deepak. Can you hold on the merging the revert.. I will check the pr
and confirm.

**Pull request merged**